File: max_sum.rb
def max_sum(numbers)
  sum_values = []
  
  numbers.each do |row|
    sum_values.push(row.inject(:+))
  end

  numbers.transpose.each do |column|
    sum_values.push(column.inject(:+))
  end
  sum_values.max
end

numbers = [[6, 7, 8], [1, 4, 29], [4, 78, 12]]
max_sum(numbers)

File: max_sum.js
const numbers = [[6, 7, 8], [1, 4, 29], [4, 78, 12]];
let numbersCol = [];
let sumNumbers = [];

maxSum = (numbers) => {
  numbers.filter(number => sumNumbers.push(number.reduce((sum, number) => sum + number)));
  numbersCol = numbers.map((column, number) => numbers.map(row => row[number]));
  numbersCol.filter(number => sumNumbers.push(number.reduce((sum, number) => sum + number)));
  return Math.max.apply(null, sumNumbers);
};

console.log(maxSum(numbers));